Don't test Content-Type for aws lambda input

omrihar opened the pull request:
This addresses an issue with AWS Lambda where sometimes a `Content-Type` header is not being sent to the Lambda function even though it is set by the client. This resolves https://github.com/bentoml/BentoML/issues/893.

<!--- Thanks for sending a pull request! Please make sure to read the contribution guidelines, then fill out the blanks below before requesting a code review. -->

## Description
<!--- Describe your changes in detail -->
<!--- Attach screenshots here if appropriate. -->
With this change the JsonInput adapter will, by default, try to parse the input as JSON without testing for the `Content-Type` header, for AWS Lambda functions. If a JSONDecodeError is raised, it will raise a BadInput error.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- If it is based on a conversation in slack channel, pls quote related messages here -->
I had problems with AWS Lambda not transferring the correct headers and as a result, a server error was raised because the Content-Type was missing. We also discussed this on Slack.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
In a function I use in production, I made this change manually to the lambda function and deployed it.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build
<!--- [ ] Others: describe the type of change if it does not fit to categories listed -->

## Component(s) if applicable
- [X ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- If you plan to update documentation or tests in follow-up, please note -->
- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

omrihar committed:
Don't test Content-Type for aws lambda input

This addresses an issue with AWS Lambda where sometimes a `Content-Type` header is not being sent to the Lambda function even though it is set by the client. This resolves https://github.com/bentoml/BentoML/issues/893.

parano commented:
Hi @omrihar - just saw this pull request from the linked issue, I think you accidentally send this request to the master branch of your fork instead of the main repo.  You can go to the main repo's pull requests page https://github.com/bentoml/BentoML/pulls, click `New pull request`, and then click the "compare cross forks" text link, which will allow you to create a pull request to send changes in your fork to the main repo.

omrihar commented:
Woops, thanks! That's what happens when I try to get stuff done quickly at night...

omrihar committed:
Adapt wording of exception to reflect reality

omrihar committed:
Update test to reflect changes in json handler

The json handler accepts any request to AWS Lambda as long as the body is valid json. 
This change reflects that in the tests.

omrihar committed:
Remove trailing whitespaces

parano committed:
Match the error output

parano committed:
Auto format code
